### PR TITLE
modify parse_json_into_columns to handle multiple paths

### PR DESCRIPTION
**Pull request by huttotw**
This adds a new field `additional_paths` where we will look in order for values if the `path` does not return a valid json value. This functions basically like COALESCE.

### CODE REVIEW

**Review by sgagofs — APPROVED** [review]
LGTM! This looks like a good change.

**Review by craigrmccown — COMMENTED** [review]
To simplify this a bit, should we use a single `paths` configuration? Or, do you think that supporting both `path` and `additional_paths` is more ergonomic because a single path is most common?

**Comment by huttotw** [comment]
I like that suggestion @craigrmccown, updated!

File: models/staging/stg_events__all.sql [hunk 1]
```diff
@@ -147,29 +191,45 @@ select
...
+                    "paths": [
+                        "$.element_definition_id",
+                        "$.click.target.named_element_id",
+                        "$.change.target.named_element_id",
+                        "$.copy.target.named_element_id",
+                        "$.form_abandon.target.named_element_id",
+                        "$.highlight.target.named_element_id",
+                        "$.paste.target.named_element_id",
+                        "$.target.element_definition_id",
+                        "$.watched_element.target.named_element_id",
```
Comment (camphillips22) [inline models/staging/stg_events__all.sql:224]: Why do we need these paths? `element_definition_id` should only show up in a single field, `$.element_definition_id`.

Comment (craigrmccown) [inline models/staging/stg_events__all.sql:224]: Are these paths correct? Are there examples of event data that contains these paths? I'm looking at the types we use to serialize this data, and I don't see these paths. Also, the event type prefix is unfamiliar to me.

**Review by camphillips22 — CHANGES_REQUESTED** [review]
Blocking this for now. While I'm not necessarily opposed to the functionality, we should never have to do this coalescing as our data model should only contain the data in a single place.

**Comment by huttotw** [comment]
> we should never have to do this coalescing as our data model should only contain the data in a single place.

Agreed. I confirmed that I was developing against a old copy of data and the issue that this PR would have solved is not actually present. Closing.

**Pull request closed without merging**